superset/db_engine_specs.py

```diff
@@ -88,6 +90,23 @@ def extra_table_metadata(cls, database, table_name, schema_name):
         """Returns engine-specific table metadata"""
         return {}
 
+    @classmethod
+    def apply_limit_to_sql(cls, sql, limit, database):
+        """Alters the SQL statement to apply a LIMIT clause"""
+        if cls.limit_method == LimitMethod.WRAP_SQL:
+            qry = (
+                select('*')
+                .select_from(
+                    TextAsFrom(text(sql), ['*']).alias('inner_qry'),
+                )
+                .limit(limit)
+            )
+            return database.compile_sqla_query(qry)
+        elif LimitMethod.FORCE_LIMIT:
+            no_limit = re.sub(r'(?i)\s+LIMIT\s+\d+;?(\s|;)*$', '', sql)
```

Comment:
Scary! :-O

BTW, you can use backreferences here to replace the LIMIT number in one pass (untested):

```python
sql = re.sub(r'(?i)(.*\s+LIMIT\s+)\d+\s*;?\s*$', r'\1 {0}'.format(limit), sql)
```

Reply:
I barely understand what I wrote here :/

Reply:
> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.

Yeah, I just realized my example does not work when there's no LIMIT clause. :-/

You know about verbose regexes? Eg:

```python
no_limit = re.sub(r"""
    (?ix)        # case insensitive, verbose
    \s+          # whitespace
    LIMIT\s+\d+  # LIMIT $ROWS
    ;?           # optional semi-colon
    (\s|;)*$     # any number of trailing whitespace or semi-colons, til the end
""", '', sql)
```

Though in this case I think it's pretty straightforward. You might wanna `rstrip` whitespace and semicolons from the end of `sql`, that would simplify the regex, no?